creator-node/src/diskManager.ts

```diff
@@ -566,3 +420,194 @@ export async function sweepSubdirectoriesInFiles(
+    do {
+      legacyStoragePathsAndCids =
+        await DbManager.getNonDirLegacyStoragePathsAndCids(cursor, BATCH_SIZE)
+      if (legacyStoragePathsAndCids.length) {
+        const copiedFilePaths = await _copyLegacyFiles(
+          legacyStoragePathsAndCids,
+          prometheusRegistry,
+          logger
+        )
+        const dbUpdateSuccessful = await DbManager.updateLegacyPathDbRows(
+          copiedFilePaths,
+          logger
+        )
+        if (dbUpdateSuccessful) {
+          await _deleteFiles(
+            copiedFilePaths.map((file) => file.legacyPath),
+            logger
+          )
+        }
+      }
+      await timeout(1000) // Avoid spamming fast queries
+    } while (legacyStoragePathsAndCids.length === BATCH_SIZE)
+  }
+}
+
+async function _migrateDirsWithLegacyStoragePaths(logger: Logger) {
+  const BATCH_SIZE = 100
+  // Paginate at each character in range [Qmz, ..., Qma, QmZ, ..., QmA, Qm9, ..., Qm0]
+  for (const char of _getCharsInRanges('az', 'AZ', '09')) {
+    const cursor = 'Qm' + char
+
+    // Query for legacy storagePaths in the pagination range until no more results are returned
+    let legacyStoragePathsAndCids: { storagePath: string; cid: string }[] = []
+    do {
+      legacyStoragePathsAndCids =
+        await DbManager.getDirLegacyStoragePathsAndCids(cursor, BATCH_SIZE)
+      if (legacyStoragePathsAndCids.length) {
+        const legacyAndNonLegacyPaths = legacyStoragePathsAndCids.map(
+          (storagePathAndCid) => {
+            return {
+              legacyPath: storagePathAndCid.storagePath,
+              nonLegacyPath: computeFilePath(storagePathAndCid.cid)
+            }
+          }
+        )
+        await DbManager.updateLegacyPathDbRows(legacyAndNonLegacyPaths, logger)
+      }
+      await timeout(1000) // Avoid spamming fast queries
+    } while (legacyStoragePathsAndCids.length === BATCH_SIZE)
+  }
+}
+
+/**
+ * For non-directory files and then later for files, this:
+ * 1. Finds rows in the Files table that have a legacy storagePath (/file_storage/<CID or dirCID>)
+ * 2. Copies the file to to the non-legacy path (/file_storage/files/<CID or dirCID>)
+ * 3. Updates the row in the Files table to reflect the new storagePath
+ * 4. Increments Prometheus metric `filesMigratedFromLegacyPath` to reflect the number of files moved
+ * @param logger logger to print errors to
+ */
+export async function migrateFilesWithLegacyStoragePaths(
+  prometheusRegistry: any,
+  logger: Logger
+): Promise<void> {
+  try {
+    await _migrateNonDirFilesWithLegacyStoragePaths(prometheusRegistry, logger)
+    await _migrateDirsWithLegacyStoragePaths(logger)
+  } catch (e: any) {
+    logger.error(`Error migrating legacy storagePaths: ${e}`)
+  }
+
+  // Keep calling this function recursively without an await so the original function scope can close
+  return migrateFilesWithLegacyStoragePaths(prometheusRegistry, logger)
```

Comment:
I think it should be a one-time job since syncs recompute storagePaths to prevent adding more paths to migrate, but just in case I pushed a change now to keep it running but slow down after 1 iteration

Reply:
makes sense